src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -177,6 +177,110 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
         }
     };
 
+    template<typename T>
+    struct ConversionTrait<std::unordered_map<std::string, T>>
+    {
+        std::unordered_map<std::string, T> FromJson(const Json::Value& json) const
+        {
+            ConversionTrait<T> trait;
+            std::unordered_map<std::string, T> val;
+
+            for (const auto& element : json.getMemberNames())
+            {
+                val[element.c_str()] = trait.FromJson(json[JsonKey(element)]);
+            }
+
+            return val;
+        }
+
+        bool CanConvert(const Json::Value& json) const
+        {
+            if (!json.isObject())
+            {
+                return false;
+            }
+            ConversionTrait<T> trait;
+            for (const auto& element : json.getMemberNames())
+            {
+                if (!trait.CanConvert(json[JsonKey(element)]))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        Json::Value ToJson(const std::unordered_map<std::string, T>& val)
+        {
+            ConversionTrait<T> trait;
+            Json::Value json{ Json::objectValue };
+
+            for (const auto& [k, v] : val)
+            {
+                json[JsonKey(k)] = trait.ToJson(v);
+            }
+
+            return json;
+        }
+
+        std::string TypeDescription() const
+        {
+            return "map";
+        }
+    };
+
+    template<typename T>
+    struct ConversionTrait<std::unordered_map<std::wstring, T>>
+    {
+        std::unordered_map<std::wstring, T> FromJson(const Json::Value& json) const
+        {
+            ConversionTrait<T> trait;
+            std::unordered_map<std::wstring, T> val;
+
+            for (const auto& element : json.getMemberNames())
+            {
+                val[til::u8u16(Detail::GetStringView(json))] = trait.FromJson(json[JsonKey(element)]);

```

Comment:
this seems wrong -- the json object you're looking at which is an object cannot be used to get a string view!

Reply:
*oops* that should be `element`, thanks for the catch! Fixed